x/sysproxy/sysproxy_linux.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux && !android
+
+package sysproxy
+
+import (
+	"errors"
+	"os/exec"
+	"strings"
+)
+
+type ProxyType string
+
+const (
+	proxyTypeHTTP  ProxyType = "http"
+	proxyTypeHTTPS ProxyType = "https"
+	proxyTypeSOCKS ProxyType = "socks"
+)
+
+func SetWebProxy(host string, port string) error {
+	// Set HTTP and HTTPS proxy settings
+	if err := setProxySettings(proxyTypeHTTP, host, port); err != nil {
+		return err
+	}
+	if err := setProxySettings(proxyTypeHTTPS, host, port); err != nil {
+		return err
+	}
+	if err := setManualMode(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ClearWebProxy() error {
+	// clear settings
+	if err := setProxySettings(proxyTypeHTTP, "127.0.0.1", "0"); err != nil {
```

Comment:
Same: can we use empty strings? It's more obvious.

Reply:
sure, I will test it first to make sure it works.

Reply:
@fortuna it seems like setting the strings to empty value has unintended consequences. For example, on macos, if I do 

```
networksetup -setwebproxy Wi-Fi ""  ""
```

and then get the setting values, the port is returned as 0 but address is empty:
```
networksetup -getwebproxy Wi-Fi     
Enabled: Yes
Server: 
Port: 0
Authenticated Proxy Enabled: 0
```

This breaks my tests since set and read-back values do not match. 

I am concerned changing this causes some stability issues if not throughly tested on all platforms. Since my tests are currently passing on all platforms and I have also tested the proxy behavior on all platforms end-to-end, let's use a default IP and port instead of empty string, if that is okay with you